### ISSUE TITLE
WebGPU frontend not working with window opacity

### ISSUE THREAD

**Issue by max-legrand** [wez/wezterm#5790]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

20240701-070926-69686f45 x86_64-pc-windows-msvc

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

When using the WebGPU front end, setting the window opacity does not work. Reverting back to OpenGL restores the transparency effect.

### Configuration

```lua
config.front_end = "WebGpu"
config.webgpu_power_preference = "HighPerformance"
config.webgpu_force_fallback_adapter = true
local gpus = wezterm.gui.enumerate_gpus()
config.webgpu_preferred_adapter = gpus[1]
config.window_background_opacity = 0.8
```

### Expected Behavior

Background should become transparent, however, it just darkens. Setting opacity to 0 results in a completely black background.

**Comment by hjm7** [comment]
For me, window opacity work when vulkan, but dx12 not work.

**Comment by wez** [comment]
FWIW, this is not a wezterm issue per-se, but due to upstream wgpu/drivers.  Changing the front end or gpu adapter selection may yield different results.

https://wezfurlong.org/wezterm/config/lua/config/front_end.html?h=front_en#webgpu

**Comment by 0xJWLabs** [comment]
@wez OpenGL works, but WebGpu font rendering is better though.

**Comment by ZacharyHM** [comment]
Indeed, I was getting terrible glitching depending on transparency and window size, so Webgpu and High_Performance was the answer to that, but I lost transparency, and I can confirm that this config worked for me, (I also had W11 and Vulkan).

```lua
local gpus = wezterm.gui.enumerate_gpus()

return {

	webgpu_preferred_adapter = gpus[1],
	front_end = "WebGpu",
	webgpu_power_preference = "HighPerformance",
```

**Comment by 0xJWLabs** [comment]
@ZacharyHM but it means we lost transparency?

**Comment by DnBaal** [comment]
Looks like transparency not working with Dx12 webgpu. Very sad , because Dx12 performs better at win11.
